capstone/src/main/java/com/google/sps/servlets/YoutubeTrendingServlet.java

```diff
@@ -36,16 +36,23 @@
   private static final String KEY = "API_KEY";
   private static final String NUM_RESULTS = "2";
   private static final MediaType JSON = MediaType.get("application/json; charset=utf-8");
+  private String postRequestBodyData;
   OkHttpClient client = new OkHttpClient();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String videoCategoryId = request.getParameter("videoCategoryId");
-    String completeUrl = BASE_URL + "&maxResults=" + NUM_RESULTS + "&regionCode=US&videoCategoryId=" + videoCategoryId + "&key=" + KEY;
+    String completeUrl = BASE_URL + "&maxResults=" + NUM_RESULTS + "&regionCode=US&videoCategoryId=" + postRequestBodyData + "&key=" + KEY;
+    System.out.println(completeUrl);
     String output = get(completeUrl);
     response.setContentType("application/json");
     response.getWriter().println(output);  
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    postRequestBodyData = request.getReader().readLine().replaceAll("^\"+|\"+$", "");
+    doGet(request, response);
```

Comment:
Same thing here - if possible, I'd recommend moving the code from doGet into this function to remove the extra server call.